containers/pytorch/training/tpu/2.5.1/transformers/4.46.3/py310/Dockerfile

```diff
@@ -0,0 +1,48 @@
+FROM us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/xla:r2.5.1_3.10_tpuvm
+# Google maintained pytorch xla image with PyTorch=2.5.1 and Python=3.10
+# Read more about it here: https://github.com/pytorch/xla?tab=readme-ov-file#docker
+
+LABEL maintainer="Hugging Face"
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Versions
+ARG OPTIMUM_TPU='0.2.2'
+ARG TRANSFORMERS='4.46.3'
+ARG PEFT='0.13.2'
+ARG TRL='0.12.1'
+ARG DATASETS='3.1.0'
+ARG ACCELERATE='1.1.0'
+ARG EVALUATE='0.4.3'
+ARG SAFETENSORS='0.4.5'
+
+# Update pip
+RUN pip install --upgrade pip
+
+# Install Hugging Face Libraries
+RUN pip install --upgrade --no-cache-dir \
+  transformers[sklearn,sentencepiece]==${TRANSFORMERS} \
+  datasets==${DATASETS} \
+  accelerate==${ACCELERATE} \
+  evaluate==${EVALUATE} \
+  peft==${PEFT} \
+  trl==${TRL} \
+  safetensors==${SAFETENSORS} \
+  jupyter notebook
+
+# Install Optimum TPU
+RUN pip install git+https://github.com/huggingface/optimum-tpu.git@v${OPTIMUM_TPU}
+# Add example
+ADD https://raw.githubusercontent.com/huggingface/optimum-tpu/v${OPTIMUM_TPU}/examples/language-modeling/gemma_tuning.ipynb \
+  /notebooks/gemma_tuning.ipynb
```

Comment:
As mentioned, not sure whether we should be shipping an example within the container here, but just in case cc @philschmid to get his thoughts too! I guess that it's a nice to have for users to be able to run it quickly, but would a simple example using a mount to include it work too? As we're going to create dedicated examples maybe this is no longer required, right?

Reply:
This is the same as the previous version https://github.com/huggingface/Google-Cloud-Containers/blob/b174d631b9106d1e869f634c1839aff2cd0d7bac/containers/pytorch/training/tpu/2.4.0/transformers/4.41.1/py310/Dockerfile#L33. But yes, let me know for the example what do you prefer and I can remove it